workflows/fetchdatairidanext.nf

```diff
@@ -62,11 +63,19 @@ workflow FETCHDATAIRIDANEXT {
             return [["id": new_id, "irida_id": meta.irida_id[0], "insdc_accession": meta.insdc_accession[0]], meta.insdc_accession[0]]
         }
 
-    FASTQ_DOWNLOAD_PREFETCH_FASTERQDUMP_SRATOOLS (
-        ch_sra_ids = input,
-        ch_dbgap_key = []
-    )
-    ch_versions = ch_versions.mix(FASTQ_DOWNLOAD_PREFETCH_FASTERQDUMP_SRATOOLS.out.versions)
+    // Determine workflow based off of the provider
+    if ( params.provider.toLowerCase() == 'sra' ) {
+        FASTQ_DOWNLOAD_PREFETCH_FASTERQDUMP_SRATOOLS (
+            ch_sra_ids = input,
+            ch_dbgap_key = []
+        )
+        ch_versions = ch_versions.mix(FASTQ_DOWNLOAD_PREFETCH_FASTERQDUMP_SRATOOLS.out.versions)
+    } else {
```

Comment:
I would probably change this to `else if ( params.provider.toLowerCase() == 'ena')` and have another `else` statement that exits with an error/throws an exception. Otherwise, anything other than `sra` is considered the same as `ena`.

Reply:
I can definitely add this! Won't it fail in the validation of the params before here though with the `enum` defining only having `SRA` and `ENA`?

Reply:
Yes. Correct. It's possible to adjust the Nextflow configuration to ignore validation though, so it's best to also do this check in the code.